src/map.js

```diff
@@ -0,0 +1,21 @@
+const map = ( array, operation ) => {
+  const fn = operation || ( () => {} )
+  let results =[]
+  if(typeof operation === 'function'){
+    for(let index = 0; index < array.length; index++ ){
+      let item = array[index]
+      let result = fn(item)
+      results.push(result)
+    }
+  }
+  else if (typeof operation === 'string'){
+    for(let index = 0; index < array.length; index++ ){
+      let item = array[index]
+      let result = item[operation]
```

Comment:
Given that this is the only difference between the two different approaches, I would look at ways to simplify this code.  Let's discuss; probably easier to do in person than via comments.  :P